code/game/objects/structures/window.dm

```diff
@@ -143,10 +143,26 @@
 	if(!can_be_reached(user))
 		return
 	user.changeNext_move(CLICK_CD_MELEE)
-	user.visible_message("<span class='notice'>[user] knocks on [src].</span>", \
-		"<span class='notice'>You knock on [src].</span>")
+	if(ishuman(user) && density)
+		var/mob/living/carbon/human/H = user
+		if((HAS_TRAIT(H, TRAIT_DUMB)) && Adjacent(user))
+			playsound(src, 'sound/effects/bang.ogg', 25, TRUE)
+			user.do_attack_animation(src, ATTACK_EFFECT_PUNCH)
+			if(!istype(H.head, /obj/item/clothing/head/helmet))
+				H.visible_message("<span class='danger'>[user] headbutts the [src].</span>", \
+									"<span class='userdanger'>You headbutt the [src]!</span>")
+				H.Paralyze(100)
```

Comment:
is this a joke 🥴

Reply:
It's actually the same check airlocks already have. >w>